### PR TITLE
docs(gateway): dynamic plugin ordering

### PR DESCRIPTION
**Pull request by jschmid1**
Signed-off-by: Joshua Schmid <jaiks@posteo.de>

### Summary

Releated PR: https://github.com/Kong/kong-ee/pull/3164

### Reason

Internal: https://konghq.atlassian.net/browse/FT-2387

### Testing

You should be able to test this feature by following the docs. If not, the docs are crap.

### Preview

https://deploy-preview-3905--kongdocs.netlify.app/

### CODE REVIEW

**Comment by jschmid1** [comment]
Not sure if this was your intention @mheap ?

image

This PR looks at 45k changed lines now :)

File: src/gateway/configure/dynamic-plugin-ordering/index.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+### Dependency tokens
+
+Use one of the following tokens to describe a dependency to a plugin:
+
+* `before`
+	 The plugin will be executed _before_ a specified plugin or list of plugins.
+
+* `after`
+	 The plugin will be executed _after_ a specified plugin or list of plugins.
+
```
Comment (lena-larionova) [inline src/gateway/configure/dynamic-plugin-ordering/index.md:26]: Not sure if my suggestion is entirely correct - can you list out multiple plugins? eg:

```
PluginA:
  ordering:
    before:
      access:
        - PluginB
        - PluginC
        - PluginD
```
Or can you only set one plugin?

Comment (jschmid1) [reply src/gateway/configure/dynamic-plugin-ordering/index.md:26]: multiple plugins can be set

File: src/gateway/configure/dynamic-plugin-ordering/get-started.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+    If you switched it to **Scoped**, the rate limiting would apply the plugin to only one Service, Route. Note that Consumer scoping is not supported
+
+    > **Note**: By default, the plugin is automatically enabled when the form is submitted. You can also toggle the **This plugin is Enabled** button at the top of the form to configure the plugin without enabling it. For this example, keep the plugin enabled.
+
+6. Scroll down and complete only the following fields with the following parameters.
+    1. config.limit: `5`
+    2. config.sync_rate: `-1`
+    3. config.window_size: `30`
+    4. ordering.. TODO: this needs to be implemented in KongManager first
```
Comment (jschmid1) [inline src/gateway/configure/dynamic-plugin-ordering/get-started.md:37]: Kong Manager integration won't happen in 3.0. We can drop the entire section.

Comment (jschmid1) [inline src/gateway/configure/dynamic-plugin-ordering/get-started.md:37]: Kong Manager integration won't happen in 3.0. We can drop the entire section.

Comment (jschmid1) [reply src/gateway/configure/dynamic-plugin-ordering/get-started.md]: see c1b359b615628c18e5639bc6307e901d2a62ed8a

Comment (jschmid1) [reply src/gateway/configure/dynamic-plugin-ordering/get-started.md]: see c1b359b615628c18e5639bc6307e901d2a62ed8a

**Review by lena-larionova — APPROVED** [review]

**Pull request merged**